plugins/org.eclipse.mat.parser/src/org/eclipse/mat/parser/internal/snapshot/IObjectMarker.java

```diff
@@ -0,0 +1,18 @@
+package org.eclipse.mat.parser.internal.snapshot;
+
+import org.eclipse.mat.SnapshotException;
+import org.eclipse.mat.snapshot.ExcludedReferencesDescriptor;
+import org.eclipse.mat.snapshot.ISnapshot;
+import org.eclipse.mat.util.IProgressListener;
+
+public interface IObjectMarker
+{
+
+    int markSingleThreaded();
+
+    void markMultiThreaded(int threads) throws InterruptedException;
+
+    int markSingleThreaded(ExcludedReferencesDescriptor[] excludeSets, ISnapshot snapshot)
```

Comment:
I think this name could be better for exposed interface. If needed we could add a wrapper layer.

Reply:
Resolving since this is internal.